lib/multiaddr.js

```diff
@@ -3,13 +3,32 @@
  * @returns {string} Parsed URI, e.g. `http://127.0.0.1:80`
  */
 export function multiaddrToHttpUrl (addr) {
-  const [, hostType, hostValue, ipProtocol, port, scheme, ...rest] = addr.split('/')
+  const [multiAddr, httpPathMultiAddr] = addr.split('/http-path')
+  const [, hostType, hostValue, ...multiAddrParts] = multiAddr.split('/')
+  let scheme, path, rest, port
+  if (addr.includes('/http-path')) {
+    [scheme, ...rest] = multiAddrParts
+    try {
+      // Remove leading slash and parse URI-encoded path
+      // If the http-path is empty, both `.../http-path/` and `.../http-path` are valid.
```

Comment:
```suggestion
```

Since we documented that we're removing leading slashes, it's implied that `'/'` and `''` are equal.